robottelo/libvirt_discovery.py

```diff
@@ -0,0 +1,197 @@
+"""Utilities to create virtual host on libvirt with and without PXE boot
+
+Hosts are virtual guests provisioned on a external ``libvirt_host``. All
+guests images are stored on the ``image_dir`` path on external libvirt
+server.
+
+Make sure to configure the ``compute_resources`` section on the configuration
+file. Also make sure that the ``vlan_networking`` section is properly
+configured.
+
+"""
+import logging
+import os
+
+from fauxfactory import gen_mac
+from robottelo import ssh
+from robottelo.config import settings
+
+logger = logging.getLogger(__name__)
+
+
+class LibvirtGuestError(Exception):
+    """Exception raised for failed virtual guests on external libvirt"""
+
+
+class LibvirtGuest(object):
```

Comment:
This code is very similar to the virtual machine one. We may consider to create a base virtual machine class that is able to handle Libvirt and snap-guest virtual machines.